### PR TITLE
Fetch data from API 

### PR DESCRIPTION
**Pull request by liyatheresa**
This PR adds the functionality of showing suggestion words based on user entry. Suggestions are fetched from [`datamuse`](https://www.datamuse.com/api/) API.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
 import "./App.css";
+
 function App() {
-  const [search, setSearch] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [searchResult, setSearchResults] = useState([]);
+  const fetchWordSearchResults = async () => {
+    try {
+      const { data: response } = await axios.get(
+        "https://api.datamuse.com/sug",
```
Comment (divinsmathew) [inline src/App.js]: Move constants into a new file `constants.js`.

File: src/Components/SearchBar/index.jsx [hunk 2]
```diff
@@ -20,10 +20,14 @@ const SearchBar = (props) => {
           placeholder="Search the word..."
           onFocus={showSuggestions}
           onBlur={hideSuggestions}
+          onChange={(e) => {
+            props.setSearchTerm(e.target.value);
+          }}
```
Comment (divinsmathew) [inline src/Components/SearchBar/index.jsx]: Scope for simplification:
```suggestion
          onChange={(e) => props.setSearchTerm(e.target.value)}
```

File: src/Components/Suggestions/index.jsx [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 
-const Suggestions = () => {
+const Suggestions = ({ searchResult }) => {
   return (
     <>
       <ul className="suggestion-list">
-        <li className="list-item">hi</li>
+        {searchResult.map((result) => (
+          <li key={result.word}>{result.word}</li>
+        ))}
```
Comment (divinsmathew) [inline src/Components/Suggestions/index.jsx:10]: Same.

**Review by divinsmathew — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 4]
```diff
@@ -1,14 +1,39 @@
...
 import SearchBar from "./Components/SearchBar";
-
+import axios from "axios";
 import "./App.css";
+
 function App() {
-  const [search, setSearch] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [searchResult, setSearchResults] = useState([]);
+  const fetchWordSearchResults = async () => {
```
Comment (divinsmathew) [inline src/App.js:12]: I would recommend adding a re-usable utility function to do web requests. The function should take in the request type, endpoint URL, and params, and should return the request result. Also, move the exception handling into this function.

File: src/Components/Suggestions/index.jsx (hunk 3 above)
Comment (liyatheresa) [reply src/Components/Suggestions/index.jsx:10]: Prettier automatically adds this parenthesis on save.

**Review by divinsmathew — APPROVED** [review]

**Pull request merged**